### PR TITLE
feat(args): add support for empty slice un/marshahaling

### PR DESCRIPTION
**Pull request by jerome-quere**

### CODE REVIEW

File: internal/args/marshal_test.go [hunk 1]
```diff
@@ -101,6 +101,17 @@ func TestMarshal(t *testing.T) {
 		},
 	}))
 
+	t.Run("empty-slice", run(TestCase{
+		data: &Slice{
+			Strings:    []string{},
+			StringsPtr: []*string{},
```
Comment (kindermoumoute) [inline internal/args/marshal_test.go]: StringsPtr should be `*[]string{}`

File: internal/args/unmarshal.go [hunk 2]
```diff
@@ -177,14 +207,19 @@ func set(dest reflect.Value, argNameWords []string, value string) error {
 
 	case reflect.Slice:
 		// If type is a slice:
-		// We check if argNameWords[0] is an number to handle cases like keys.0.value=12
+
+		// We handle special case where array=none creates an empty slice.
```
Comment (QuentinBrosse) [inline internal/args/unmarshal.go]: nit: `When array=none, creates an empty slice.`

File: internal/args/marshal_test.go [hunk 3]
```diff
@@ -101,6 +101,17 @@ func TestMarshal(t *testing.T) {
...
+		},
+		expected: []string{
+			"strings=none",
```
[the remark this answers is not in the document]
Comment (jerome-quere) [reply internal/args/marshal_test.go]: not extactly the same
`absent` => `[]string(nil)`
`=none` => `[]string{}`

[the remark this answers is not in the document]
Comment (kindermoumoute) [reply internal/args/marshal_test.go]: AFAIK we don't have any use case for this as we always handle `[]string(nil) == []string{}` in our APIs

File: internal/args/marshal_test.go [hunk 4]
```diff
@@ -104,11 +110,13 @@ func TestMarshal(t *testing.T) {
 	t.Run("empty-slice", run(TestCase{
 		data: &Slice{
 			Strings:    []string{},
+			SlicePtr:   &emptySlicePtr,
 			StringsPtr: []*string{},
```
Comment (kindermoumoute) [inline internal/args/marshal_test.go]: IMO we shouldn't keep `[]*string{}` in this test

**Pull request merged**